service.js

```diff
@@ -80,6 +81,7 @@ ServiceSpec.prototype.compile = function process(def, spec) {
     for (var index = 0; index < def.functions.length; index++) {
         self.compileFunction(def.functions[index], spec);
     }
+    spec[self.name] = self.functionsByName;
```

Comment:
Everything else in this PR looks sensible, but this line seems off to me:
- why not just make them use `spec.services.Foo.functionsByName` directly?
- could even provide a `spec.getFunction(service, name) -> ?functionSpec` convenience
- finally, if we must have this contract, can we invert the control so that Spec mutates itself, and not compile here?  At worst it seems like this should be part of `ServiceSpec#link`, not `#compile`